libecs/source/Core/Component/manager.cxx

```diff
@@ -44,4 +44,10 @@ void Manager::EntityDestroyed(Entity entity)
 {
     for (auto &componentArray: m_componentArrays) { componentArray->setValueForEntity(entity, std::nullopt); }
 }
+
+std::optional<std::reference_wrapper<IComponentArray>> Manager::GetComponentArray(ComponentId index)
+{
+    return *m_componentArrays.at(index);
+}
```

Comment:
Why returning an optional ?

Two solution:
keep the optional
```suggestion
std::optional<std::reference_wrapper<IComponentArray>> Manager::getComponentArray(ComponentId index)
{
    auto idx = m_componentArrays.find(index);
    if (idx == m_componentArrays.end()) return std::nullopt;
    return *idx;
}
```

or

remove the optional (will throw if index not in m_componentArrays)
```suggestion
std::unique_ptr<IComponentArray> &Manager::getComponentArray(ComponentId index)
{
    return m_componentArrays.at(index);
}
```